example/gmres/gmres.hpp

```diff
@@ -42,28 +42,35 @@
 //@HEADER
 */
 
-#include<math.h>
-#include"KokkosKernels_IOUtils.hpp"
-#include<Kokkos_Core.hpp>
-#include<Kokkos_Random.hpp>
-#include<KokkosBlas.hpp>
-#include<KokkosBlas3_trsm.hpp>
-#include<KokkosSparse_spmv.hpp>
-#include<KokkosSparse_Preconditioner.hpp>
+#include <math.h>
+#include "KokkosKernels_IOUtils.hpp"
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+#include <KokkosBlas.hpp>
+#include <KokkosBlas3_trsm.hpp>
+#include <KokkosSparse_spmv.hpp>
+#include <KokkosSparse_Preconditioner.hpp>
 
 ////////////////////////////////////////////////////////////////////////////////
 // libstdc++ half_t overloads
 ////////////////////////////////////////////////////////////////////////////////
 #if !KOKKOS_HALF_T_IS_FLOAT
-Kokkos::Experimental::half_t abs(Kokkos::Experimental::half_t arg) {
+Kokkos::Experimental::half_t fabs(Kokkos::Experimental::half_t arg) {
   return arg < 0.0 ? -arg : arg;
 }
 
-Kokkos::complex<Kokkos::Experimental::half_t> abs(Kokkos::complex<Kokkos::Experimental::half_t> arg) noexcept {
-  return Kokkos::complex<Kokkos::Experimental::half_t>(abs(Kokkos::complex<double>((double) arg.real(), (double) arg.imag())));
+Kokkos::complex<Kokkos::Experimental::half_t> fabs(
+    Kokkos::complex<Kokkos::Experimental::half_t> arg) noexcept {
+  return Kokkos::complex<Kokkos::Experimental::half_t>(
+      fabs(Kokkos::complex<double>((double)arg.real(), (double)arg.imag())));
 }
 #endif  // KOKKOS_HALF_T_IS_FLOAT
 
+template <class T>
+KOKKOS_INLINE_FUNCTION T fabs(const Kokkos::complex<T> &x) {
```

Comment:
This is already defined in Kokkos_MathematicalFunctions, we should use that implementation.

Reply:
I do not see it. Can you point me to it?

Reply:
Yes, it's actually in Kokkos_Complex.hpp on line 717, the complex specializations are not implemented in Kokkos_MathematicalFunctions to avoid circular dependency I believe.